### DISCUSSION TITLE
No local interfaces?

### DISCUSSION THREAD

**Discussion by ZeChArtiahSaher** [reworkd/AgentGPT#1440] [Q&A]
I understand there may be fear with some level of compliance or volatility of open source, but have you seen how well latest current local models perform? Why not have options for interfacing with thigns like lm studio api for example or any type open source api for locally run llms. 

Also why not have some kind of form of search without having to query paid apis?

**Comment by yf007** [comment]
I hope someone here will pick up the glove and give us a connection to this awesome software lm studio

https://lmstudio.ai/ 

That way you won't have to pay more to openai

This is how it looks in their GUI:

# Load a model, start the server, and run this example in your terminal
# Choose between streaming and non-streaming mode by setting the "stream" field

curl http://localhost:1234/v1/chat/completions \
-H "Content-Type: application/json" \
-d '{ 
  "messages": [ 
    { "role": "system", "content": "Always answer in rhymes." },
    { "role": "user", "content": "Introduce yourself." }
  ], 
  "temperature": 0.7, 
  "max_tokens": -1,
  "stream": false
}'
